### ISSUE TITLE
PP doesn't work with FlexAttention

### ISSUE THREAD

**Issue by tianyu-l** [pytorch/torchtitan#1612]
Today PP doesn't work with FlexAttention block causal masking, because PP can't receive `eos_id` as a non-Tensor input (nor can it receive a mask function).
https://github.com/pytorch/torchtitan/blob/main/torchtitan/train.py#L433

This regression is coming from a recent refactor https://github.com/pytorch/torchtitan/pull/1424 to move `eos_id` out of `ModelArgs`, to remove dependency from model to tokenizer.

This is blocking optimizations from https://github.com/pytorch/torchtitan/pull/1610.
